axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/bankreconciliation/BankReconciliationService.java

```diff
@@ -243,11 +268,13 @@ protected MoveLine generateMoveLine(
     String description = "";
     description =
         description.concat(bankReconciliationLine.getBankStatementLine().getDescription());
-    if (description.length() > DESCRIPTION_SIZE_LIMIT)
+    if (description.length() > DESCRIPTION_SIZE_LIMIT) {
       description = description.substring(0, DESCRIPTION_SIZE_LIMIT - 1);
+    }
```

Comment:
Need to call the existing AOS method to limit the description content. https://github.com/axelor/axelor-open-suite/blob/master/axelor-tool/src/main/java/com/axelor/apps/tool/StringTool.java#L295
Also, you should truncate after you have compute the description. else, if the field bankReconciliationLine.getReference() take more 16 char you get an error.